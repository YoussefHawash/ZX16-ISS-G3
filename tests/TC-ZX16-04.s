.text
.org 0x0020

main:
    # Initialize SP
    li16 x2, 0xEFFE        # x2 = stack pointer

    # Store word
    li16 x3, 0x1234
    sw    x3, -2(x2)       # SP-2 = 0xEFFC
    lw    x4, -2(x2)       # x4 = 0x1234

    # Store signed byte
    li    x5, -12
    sb    x5, -3(x2)       # SP-3 = 0xEFFB
    lb    x6, -3(x2)       # x6 = -12

    # Store unsigned byte
    li16    x7, 0x3F
    sb    x7, -4(x2)
    lbu   x1, -4(x2)       # x1 = 63