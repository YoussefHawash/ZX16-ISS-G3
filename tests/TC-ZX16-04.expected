ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
LUI x2, 0x1e0
ADDI x2, 0x7e
LUI x3, 0x024
ADDI x3, 0x34
SW x3, -2(x2)
LW x4, -2(x2)
LI x5, 0x74
SB x5, -3(x2)
LB x6, -3(x2)
LUI x7, 0x000
ADDI x7, 0x3f
SB x7, -4(x2)
LBU x1, -4(x2)
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0